pdns/recursordist/mtasker_fcontext.cc

```diff
@@ -32,6 +32,15 @@ using boost::context::make_fcontext;
 using boost::context::detail::make_fcontext;
 #endif /* BOOST_VERSION < 106100 */
 
+// __CET__ is set by the compiler if relevant, so far only relevant/tested for amd64 on OpenBSD
+#if defined(__amd64__)
+#if __CET__ & 0x1
+#define CET_ENDBR __asm("endbr64")
```

Comment:
Would it perhaps make sense to include `<cet.h>` and use `_CET_ENDBR`, which is set to `endbr64`, `endbr32` or nothing (when `__CET__ & 0x1` is `false`)?

Reply:
Ah, too bad!

Reply:
Do we care about the 32-bit case for the recursor?

Reply:
> Do we care about the 32-bit case for the recursor?

AFAIK OpenBSD/i386 does not do enforcement. I might add the 32-bit case, but I hardly think it matters at this moment, and likely never will.

Reply:
Makes sense to me, let's leave it out for now then.